examples/checkpoint_example_2.cc

```diff
@@ -42,33 +42,92 @@
 //@HEADER
 */
 
+/// [Serialize custom structure]
+
 #include <checkpoint/checkpoint.h>
 
 #include <cstdio>
 
 namespace checkpoint { namespace examples {
 
+// \struct MyTest2
+// \brief Simple structure with one variable of built-in type
+struct MyTest2 {
+  int c = 41;
+
+  // \brief Default constructor
+  //
+  // The default constructor is needed for the (de)serialization.
+  // (required for serialization)
+  MyTest2() = default;
+
+  // \brief Templated function for serializing/deserializing
+  // a variable of type `MyTest2`
+  //
+  // \tparam <Serializer> { Type for storing the serialized result }
+  // \param[in,out] Variable for storing the serialized result
+  // 
+  // \note The routine `serialize` is actually a two-way routine:
+  // - it creates the serialized result `s` by copying
+  //   the variable `c`; this creation phase is run
+  //   when the status `s.isPacking()` is `true`.
```

Comment:
it could be better to refer to serialization mode directly (`Packing`, `Unpacking` as defined in `base_serializer.h`) instead of `s.is...()` status

Reply:
Fixed.